### PR TITLE
Make handling fake sup/sub optional

### PR DESCRIPTION
**Pull request by kylegibson**
Move the sup/sub handling in `parse_r` to a mixin that can be used optionally, instead of by default.

This issue also splits up `tests.export.test_html` into multiple separate test modules in `tests.export.html`

### CODE REVIEW

File: tests/export/html/test_direct_formatting_bold_property.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
+# coding: utf-8
```
Comment (kylegibson) [inline tests/export/html/test_direct_formatting_bold_property.py]: Nothing was changed in these tests, they were moved from `test_html`

File: tests/export/html/test_drawing_graphic_blip.py [hunk 2]
```diff
@@ -0,0 +1,344 @@
+# coding: utf-8
```
Comment (kylegibson) [inline tests/export/html/test_drawing_graphic_blip.py]: Nothing was changed in these tests, they were moved from `test_html`

File: tests/export/html/test_footnote.py [hunk 3]
```diff
@@ -0,0 +1,257 @@
+# coding: utf-8
```
Comment (kylegibson) [inline tests/export/html/test_footnote.py]: Nothing was changed in these tests, they were moved from `test_html`

File: tests/export/html/test_heading.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
+# coding: utf-8
```
Comment (kylegibson) [inline tests/export/html/test_heading.py]: Nothing was changed in these tests, they were moved from `test_html`

File: tests/export/html/test_hyperlink.py [hunk 5]
```diff
@@ -0,0 +1,139 @@
+# coding: utf-8
```
Comment (kylegibson) [inline tests/export/html/test_hyperlink.py]: Nothing was changed in these tests, they were moved from `test_html`

File: tests/export/html/test_numbering.py [hunk 6]
```diff
@@ -0,0 +1,49 @@
+# coding: utf-8
```
Comment (kylegibson) [inline tests/export/html/test_numbering.py]: Nothing was changed in these tests, they were moved from `test_html`

File: tests/export/html/test_page_break.py [hunk 7]
```diff
@@ -0,0 +1,80 @@
+# coding: utf-8
```
Comment (kylegibson) [inline tests/export/html/test_page_break.py]: Nothing was changed in these tests, they were moved from `test_html`

File: tests/export/html/test_paragraph.py [hunk 8]
```diff
@@ -0,0 +1,74 @@
+# coding: utf-8
```
Comment (kylegibson) [inline tests/export/html/test_paragraph.py]: Nothing was changed in these tests, they were moved from `test_html`

File: tests/export/html/test_property_hierarchy.py [hunk 9]
```diff
@@ -0,0 +1,240 @@
+# coding: utf-8
```
Comment (kylegibson) [inline tests/export/html/test_property_hierarchy.py]: Nothing was changed in these tests, they were moved from `test_html`

File: tests/export/html/test_style_based_on.py [hunk 10]
```diff
@@ -0,0 +1,164 @@
+# coding: utf-8
```
Comment (kylegibson) [inline tests/export/html/test_style_based_on.py]: Nothing was changed in these tests, they were moved from `test_html`

File: tests/export/html/test_xml_vulnerabilities.py [hunk 11]
```diff
@@ -0,0 +1,83 @@
+# coding: utf-8
```
Comment (kylegibson) [inline tests/export/html/test_xml_vulnerabilities.py]: Nothing was changed in these tests, they were moved from `test_html`

File: tests/export/mixins/test_faked_superscript_and_subscript.py [hunk 12]
```diff
@@ -0,0 +1,345 @@
+# coding: utf-8
```
Comment (kylegibson) [inline tests/export/mixins/test_faked_superscript_and_subscript.py]: These tests were moved from `test_html` and also `test_docx`

File: pydocx/export/mixins/faked_superscript_and_subscript.py [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+            return self.styles_manager.get_resolved_properties_for_element(
+                copied_el,
+                stack,
+            )
+
+        paragraph_properties = get_properties_with_no_font_size()
+
+        # If paragraph_properties is None then the size was not set
+        # (meaning it can't be bigger or smaller than the default for the
+        # paragraph, so early exit.
```
Comment (michaelgzoller) [inline pydocx/export/mixins/faked_superscript_and_subscript.py]: You didn't close the parens!! Chaos!

**Comment by michaelgzoller** [comment]
Looks good, @kylegibson

File: pydocx/export/mixins/faked_superscript_and_subscript.py (hunk 13 above)
Comment (kylegibson) [inline pydocx/export/mixins/faked_superscript_and_subscript.py]: I copied that comment from the original code and didn't notice the missing parenthesis :(. )) (there)

**Pull request merged**